ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/DistributedIdGenerator.java

```diff
@@ -0,0 +1,288 @@
+package io.appform.ranger.discovery.bundle.id;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import dev.failsafe.Failsafe;
+import dev.failsafe.FailsafeExecutor;
+import dev.failsafe.RetryPolicy;
+import io.appform.ranger.discovery.bundle.id.constraints.KeyValidationConstraint;
+import io.appform.ranger.discovery.bundle.id.formatter.IdFormatter;
+import io.appform.ranger.discovery.bundle.id.formatter.IdFormatters;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+/**
+ * Distributed Id Generation
+ */
+@SuppressWarnings("unused")
+@Slf4j
+public class DistributedIdGenerator {
+
+    private static final int MINIMUM_ID_LENGTH = 22;
+    protected static final SecureRandom SECURE_RANDOM = new SecureRandom(Long.toBinaryString(System.currentTimeMillis()).getBytes());
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormat.forPattern("yyMMddHHmmss");
+    private final RetryPolicy<Integer> RETRY_POLICY = RetryPolicy.<Integer>builder()
+            .withMaxAttempts(readRetryCount())
+            .handleIf(throwable -> true)
+            .handleResultIf(Objects::isNull)
+            .build();
+    protected final FailsafeExecutor<Integer> RETRIER = Failsafe.with(Collections.singletonList(RETRY_POLICY));
+    private static final Pattern PATTERN = Pattern.compile("(.*)([0-9]{12})([0-9]{4})([0-9]{6})");
+    private static final List<KeyValidationConstraint> GLOBAL_CONSTRAINTS = new ArrayList<>();
+    private static final Map<String, List<KeyValidationConstraint>> DOMAIN_SPECIFIC_CONSTRAINTS = new HashMap<>();
+    protected static final int NODE_ID = IdGenerator.getNodeId();
+    @Getter
+    private final Map<String, Map<Long, PartitionIdTracker>> dataStore = new ConcurrentHashMap<>();
```

Comment:
Maybe rename it to IdPartitionsMapping or IdPartitionStore, datastore doesn't capture what this is storing.